rules/file-contents.js

```diff
@@ -32,9 +32,9 @@ function getContext(matchedLine, regexMatch, contextLength) {
  * @returns {Promise<Result>} The lint rule result
  * @ignore
  */
-async function fileContents(fs, options, not = false) {
+async function fileContents(fs, options, not = false, any = false) {
   // support legacy configuration keys
-  const fileList = options.globsAll || options.files
+  const fileList = (any ? options.globsAny : options.globsAll) || options.files
```

Comment:
Let's not change the `file-contents` function parameter, keep the function always take `globsAll` and it matches the intention of `if all of the files returned by the globs match the supplied string, or if no files are returned.`.

You can move this part to `any-file-contents.js` instead.